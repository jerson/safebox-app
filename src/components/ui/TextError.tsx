import React from 'react';
import { StyleSheet, Text, TextProps, TextStyle } from 'react-native';
import Font from '../../modules/resources/Font';
import Colors from '../../modules/constants/Colors';

const styles = StyleSheet.create({
  text: {
    ...Font(),
    color: Colors.danger,
    fontSize: 13
  } as TextStyle,
  centerText: {
    textAlign: 'center',
    alignSelf: 'center'
  } as TextStyle
});

export interface TextErrorProps extends TextProps {
  children?: string | React.ReactNode;
  decoration?: boolean;
  center?: boolean;
}

export interface TextErrorState {}

export default class TextError extends React.Component<
  TextErrorProps,
  TextErrorState
> {
  static defaultProps = {
    decoration: true,
    center: false
  };

  render() {
    const { children, center, decoration, style, ...props } = this.props;

    return (
      <Text
        style={[styles.text, center && styles.centerText, style]}
        {...props}
      >
        {children}
        {decoration && '.'}
      </Text>
    );
  }
}
