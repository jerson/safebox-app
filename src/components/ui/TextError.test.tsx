import React from 'react';
import * as renderer from 'react-test-renderer';
import TextError, { TextErrorProps } from './TextError';

const props: TextErrorProps = { children: 'test' };

test('renders without crashing', () => {
  const snapshot = renderer.create(<TextError {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<TextError {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
