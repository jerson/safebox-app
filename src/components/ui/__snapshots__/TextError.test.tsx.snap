// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Nunito",
        "fontSize": undefined,
        "fontStyle": "normal",
        "fontWeight": "400",
      },
      Object {},
      Array [
        Object {
          "color": "#e91e63",
          "fontSize": 13,
        },
        false,
        undefined,
      ],
    ]
  }
>
  test
  .
</Text>
`;
